collectd/_service.sls

```diff
@@ -129,6 +129,36 @@
 
 {%- endif %}
 
+{%- set java_lib = salt['cmd.run']('type java >/dev/null 2>&1 && find /usr/lib/jvm -name libjvm.so -type f') %}
+{%- if java_lib|length > 0 %}
+  {%- set java_path = '/'.join(java_lib.split('/')[:-1]) %}
+  {%- set ld_is_set = salt['cmd.run']('grep "^export LD_LIBRARY_PATH" ' ~ client.defaults_file ~ ' 2>/dev/null') %}
```

Comment:
Are you aware of cases where `export LD_LIBRARY_PATH` would already be set in the defaults file? This is to deal with the case where the `{{ client.defaults_file }}` state was already applied?